### ISSUE TITLE
[Bug]: double 'afterEach' log

### ISSUE THREAD

**Issue by edwinheij** [pestphp/pest#1230]
### What Happened

In ```Pest.php``` I added an ```beforeEach``` and ```afterEach``` method like this:
```php
pest()->extend(Tests\TestCase::class)
    ->beforeEach(function () {
        $shortName = (new ReflectionClass($this))->getShortName();
        Log::debug('> Test started @ ' . $shortName . '::' . $this->__description);
    })
    ->afterEach(function () {
        $shortName = (new ReflectionClass($this))->getShortName();
        Log::debug('> Test ended @ ' . $shortName . '::' . $this->__description);
    })
    ->in('Feature');
```
In my log there are two lines for the 'Test ended':
```
[2024-09-10 18:22:51] testing.DEBUG: > Test started @ ExampleTest::it returns a successful response
[2024-09-10 18:22:51] testing.DEBUG: > Test ended @ ExampleTest::it returns a successful response
[2024-09-10 18:22:51] testing.DEBUG: > Test ended @ ExampleTest::it returns a successful response
```

All tested with a fresh Laravel installation. Is this standard behaviour of maybe a bug/feature?

### How to Reproduce

See the ```what Happend``` section

### Pest Version

3.0.2

### PHP Version

8.2.22

### Operation System

macOS

**Comment by nunomaduro** [comment]
on this now...

**Comment by nunomaduro** [comment]
fixed on `v3.0.4`.
